cmd/deploy.go

```diff
@@ -181,36 +181,41 @@ func autofillApimodel(dc *deployCmd) {
 	if !useManagedIdentity {
 		spp := dc.containerService.Properties.ServicePrincipalProfile
 		if spp != nil && !(spp.ClientID == "" && spp.Secret == "") {
-			log.Fatal("apimodel invalid: ServicePrincipalProfile is missing either the clientid or secret.")
-		}
+			log.Info("using user supplied ServicePrincipalProfile in apimodel.")
 
-		log.Warnln("apimodel: ServicePrincipalProfile was empty, creating application...")
+			dc.containerService.Properties.ServicePrincipalProfile = &api.ServicePrincipalProfile{
```

Comment:
I don't think this is needed. Should be sufficient to just fix the condition checks.